presto-native-execution/presto_cpp/main/tests/DynamicFunctionRegistererTest.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <fstream>
+#include "presto_cpp/main/dynamic_registry/DynamicFunctionRegistrar.h"
+#include "presto_cpp/main/dynamic_registry/DynamicLibraryLoader.h"
+#include "velox/common/base/Fs.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/exec/tests/utils/TempFilePath.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/functions/tests/RegistryTestUtil.h"
+#include "velox/type/SimpleFunctionApi.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+#ifndef DYNAMIC_LIB_FILE_EXT
+#error DYNAMIC_LIB_FILE_EXT must be defined in CMAKE
+#endif
+const char* kFileExtension = DYNAMIC_LIB_FILE_EXT;
+
+using namespace facebook::velox;
+namespace facebook::presto::test {
+namespace {
+class DynamicFunctionRegistererTest : public testing::Test {
+ public:
+  template <typename T>
+  struct TestDynamicFunctionMock {
+    VELOX_DEFINE_FUNCTION_TYPES(T);
+    FOLLY_ALWAYS_INLINE bool call(out_type<bool>& result) {
+      result = "123";
+      return true;
+    }
+  };
+  template <typename T>
+  struct TestDynamicFunctionMock2 {
+    VELOX_DEFINE_FUNCTION_TYPES(T);
+    FOLLY_ALWAYS_INLINE bool call(
+        out_type<Varchar>& result,
+        const arg_type<Varchar>& in) {
+      result = in;
+      return true;
+    }
+  };
+  void UdfBoolDynamicFnRegistration(
+      std::string fn_name,
+      std::string nameSpace = "") {
+    facebook::presto::registerPrestoFunction<TestDynamicFunctionMock, bool>(
+        fn_name.c_str(), nameSpace.c_str());
+  }
+  void UdfVarcharDynamicFnRegistration(
```

Comment:
Nit : Use lowercase for the first letter. This is supposed to follow camel-case format.